Check `livetest.test_idle` failure with OVH IMAP account

Using an OVH e-mail account (server ssl0.ovh.net) with livetest.py, I get one failure:
```
======================================================================
FAIL: test_idle (__main__.TestGeneral)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/data/workspace/imapclientdev/imapclient/imapclient/livetest.py", line 445, in test_idle
    self.assertIn((1, b'EXISTS'), responses)
AssertionError: (1, b'EXISTS') not found in []

----------------------------------------------------------------------
```

Since getting accounts on this provider is not free (I believe you need a paid plan or owning a domain name on their services), please assign me to this issue. I will look more closely what is happening.

So I tweak the `timeout` parameter of `idle_check` in the test to see of things goes. With `timeout=30` and multiple executions I get this (I stripped tests description/trace to ease reading):

```
$ cmd="python -m imapclient.livetest livetest-ovh.ini TestGeneral.test_idle"; for i in $(seq 6); do $cmd; done

1. -------------------------------------------------------------------
  File "imapclient/imapclient/livetest.py", line 446, in test_idle
    self.assertIn((1, b'EXISTS'), responses)
AssertionError: (1, b'EXISTS') not found in [(b'OK', b'Still here')]

Ran 1 test in 9.391s

2. -------------------------------------------------------------------
  File "imapclient/imapclient/livetest.py", line 446, in test_idle
    self.assertIn((1, b'EXISTS'), responses)
AssertionError: (1, b'EXISTS') not found in [(b'OK', b'Still here')]

Ran 1 test in 29.409s

3. -------------------------------------------------------------------
  File "imapclient/imapclient/livetest.py", line 465, in test_idle
    self.assertIn((2, b'EXISTS'), responses)
AssertionError: (2, b'EXISTS') not found in []

Ran 1 test in 32.553s


4. -------------------------------------------------------------------
  File "imapclient/imapclient/livetest.py", line 446, in test_idle
    self.assertIn((1, b'EXISTS'), responses)
AssertionError: (1, b'EXISTS') not found in [(b'OK', b'Still here')]

Ran 1 test in 27.453s


5. -------------------------------------------------------------------
  File "imapclient/imapclient/livetest.py", line 446, in test_idle
    self.assertIn((1, b'EXISTS'), responses)
AssertionError: (1, b'EXISTS') not found in [(b'OK', b'Still here')]

Ran 1 test in 29.470s


6. -------------------------------------------------------------------
  File "imapclient/imapclient/livetest.py", line 446, in test_idle
    self.assertIn((1, b'EXISTS'), responses)
AssertionError: (1, b'EXISTS') not found in [(b'OK', b'Still here')]

Ran 1 test in 29.398s
```

Server response time is a bit random and I don't even know if it is the `append` that is taking time  to be registered internally or if `idle` is sort of "lagging". The third test seems weird too, since it pass the first assert in 32.5 sec, which is rougly equal to the 2s sleep before idle_check and then 30s of timeout. 

That's all for now. Will investigate more later.

--- a/livetest.py
+++ b/livetest.py
@@ -467,7 +467,16 @@ class TestGeneral(_TestBase):
             client2.append(self.base_folder, SIMPLE_MESSAGE)
 
             # Check for the idle data
-            responses = self.client.idle_check(timeout=5)
+            # Notification can take a while to arrive. In the meantime we may
+            # receive some keep alive like (b'OK', b'Still here') that
+            # maintain the TCP connection opened.
+            start_time = time.time()
+            responses = []
+            while time.time() - start_time < 60:
+                responses = self.client.idle_check(timeout=10)
+                if (1, b'EXISTS') in responses:
+                    break
+
         finally:
             text, more_responses = self.client.idle_done()
         self.assertIn((1, b'EXISTS'), responses)
@@ -476,19 +485,19 @@ class TestGeneral(_TestBase):
         self.assertTrue(isinstance(more_responses, list))
 
         # Check for IDLE data returned by idle_done()
-
-        # Gmail now delays updates following APPEND making this
-        # part of the test impractical.
-        if self.is_gmail():
-            return
-
         self.client.idle()
         try:
             client2.select_folder(self.base_folder)
             client2.append(self.base_folder, SIMPLE_MESSAGE)
-            time.sleep(2)    # Allow some time for the IDLE response to be sent
+            time.sleep(10)  # Allow some time for the IDLE response to be sent
         finally:
             text, responses = self.client.idle_done()
+
+        if not responses:
+            # The append was not yet picked by the other connection.
+            # This happens with some servers.
+            return
+
         self.assertIn((2, b'EXISTS'), responses)
         self.assertTrue(isinstance(text, binary_type))
         self.assertGreater(len(text), 0)

